modules/ppcp-wc-gateway/src/Helper/PayUponInvoiceProductStatus.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Manage the Seller status.
+ *
+ * @package WooCommerce\PayPalCommerce\WcGateway\Helper
+ */
+
+declare( strict_types=1 );
+
+namespace WooCommerce\PayPalCommerce\WcGateway\Helper;
+
+use WooCommerce\PayPalCommerce\ApiClient\Endpoint\PartnersEndpoint;
+use WooCommerce\PayPalCommerce\ApiClient\Entity\SellerStatusProduct;
+use WooCommerce\PayPalCommerce\ApiClient\Exception\RuntimeException;
+use WooCommerce\PayPalCommerce\WcGateway\Settings\Settings;
+use WooCommerce\PayPalCommerce\WcGateway\Exception\NotFoundException;
+
+/**
+ * Class DccProductStatus
+ */
+class PayUponInvoiceProductStatus {
+
+	/**
+	 * Caches the status for the current load.
+	 *
+	 * @var bool|null
+	 */
+	private $current_status_cache;
+	/**
+	 * The settings.
+	 *
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * The partners endpoint.
+	 *
+	 * @var PartnersEndpoint
+	 */
+	private $partners_endpoint;
+
+	/**
+	 * DccProductStatus constructor.
+	 *
+	 * @param Settings         $settings The Settings.
+	 * @param PartnersEndpoint $partners_endpoint The Partner Endpoint.
+	 */
+	public function __construct(
+		Settings $settings,
+		PartnersEndpoint $partners_endpoint
+	) {
+		$this->settings          = $settings;
+		$this->partners_endpoint = $partners_endpoint;
+	}
+
+	/**
+	 * Whether the active/subscribed products support PUI.
+	 *
+	 * @return bool
+	 * @throws NotFoundException Should a setting not be found.
```

Comment:
Probably should be "When a setting not found".
But actually I don't see it thrown anywhere? If it is because of `$this->settings->get`, then it should not be possible after the `has` check before (only if it's some kind of bug). So this `@throw` should be removed.